packages/lambda-shared/src/aws/tile.metadata.imagery.ts

```diff
@@ -1,4 +1,23 @@
-import { TileMetadataTable, TileMetadataImageryRecord, TileMetadataSetRecord } from './tile.metadata';
+import {
+    TileMetadataImageryRecord,
+    TileMetadataSetRecord,
+    TileMetadataTable,
+    TileSetRuleImagery,
+} from './tile.metadata';
+
+function compareImageSets(a: TileSetRuleImagery, b: TileSetRuleImagery): number {
+    let diff = a.rule.priority - b.rule.priority;
```

Comment:
I find this quite hard to reason about, I know some people like the single return but without reading this entire function you have no idea what diff is going to be.

To me this is much more understandable
https://github.com/linz/basemaps/commit/c7ba7993e1544f7d120f7612d17b6f427549d716#diff-08397e95c4ab9c39c3653125000cfb3eR13

And has the helpful addition of comments explaining why the comparisons are being made.